src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -273,21 +279,32 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                 return false;
             }
 
-            // If this looks like an event handler don't flag such cases.
-            // However, we do want to consider EventRaise accessor as a candidate
-            // so we can flag the associated event if none of it's accessors need instance reference.
-            if (methodSymbol.HasEventHandlerSignature(wellKnownTypeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs)) &&
-                methodSymbol.MethodKind != MethodKind.EventRaise)
+            if (IsEventHandlerLike(methodSymbol, wellKnownTypeProvider) ||
+                IsExplicitlyVisibleFromCom(methodSymbol, wellKnownTypeProvider))
             {
                 return false;
             }
 
-            if (IsExplicitlyVisibleFromCom(methodSymbol, wellKnownTypeProvider))
+            var hasCorrectVisibility = blockStartContext.Options.MatchesConfiguredVisibility(Rule, methodSymbol, wellKnownTypeProvider.Compilation,
+                defaultRequiredVisibility: SymbolVisibilityGroup.All);
+            if (!hasCorrectVisibility)
             {
                 return false;
             }
 
-            var hasCorrectVisibility = blockStartContext.Options.MatchesConfiguredVisibility(Rule, methodSymbol, wellKnownTypeProvider.Compilation,
+            return true;
+        }
+
+        private static bool ShouldAnalyzeLocalFunction(IMethodSymbol methodSymbol, WellKnownTypeProvider wellKnownTypeProvider, OperationAnalysisContext context)
```

Comment:
I started from the `ShouldAnalyze` function and removed all cases that can't (or at least I think) be applied to local functions.